packages/amplify_core/lib/src/types/api/hub/api_subscription_hub_event.dart

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template amplify_common.hub.api_network_state}
+/// Network states for graphql subscriptions.
+/// {@endtemplate}
+enum NetworkState {
+  /// {@macro amplify_common.hub.api_network_state_connected}
+  connected,
+
+  /// {@macro amplify_common.hub.api_network_state_disconnected}
+  disconnected,
+
+  /// {@macro amplify_common.hub.api_network_state_failed}
+  failed;
+}
+
+/// {@template amplify_common.hub.api_intended_state}
+/// The intended network states for graphql subscriptions.
+/// {@endtemplate}
+enum IntendedState {
+  /// {@macro amplify_common.hub.api_intended_state_connected}
+  connected(),
+
+  /// {@macro amplify_common.hub.api_intended_state_disconnected}
+  disconnected();
+}
+
+/// {@template amplify_common.hub.api_subscription_status}
+/// A consolidated subscription connection status determined by the network and
+/// intended states.
+/// {@endtemplate}
+enum SubscriptionStatus {
+  /// {@macro amplify_common.hub.api_subscription_status_connected}
+  connected(),
+
+  /// {@macro amplify_common.hub.api_subscription_status_disconnected}
+  disconnected(),
+
+  /// {@macro amplify_common.hub.api_subscription_status_connected}
+  connecting(),
+
+  /// {@macro amplify_common.hub.api_subscription_status_disconnected}
+  pendingDisconnected(),
+
+  /// {@macro amplify_common.hub.api_subscription_status_failed}
+  failed();
+}
+
+class SubscriptionDetails with AWSEquatable<SubscriptionDetails> {
+  /// {@macro amplify_common.hub.api_network_state}
+  final NetworkState networkState;
+
+  /// {@macro amplify_common.hub.api_intended_state}
+  final IntendedState intendedState;
+
+  SubscriptionDetails(this.networkState, this.intendedState);
+
+  @override
+  List<Object?> get props => [intendedState, networkState];
+
+  Map<String, String> toJson() => {
+        'networkState': networkState.name,
+        'intendedState': intendedState.name,
+      };
+}
+
+class SubscriptionHubEvent extends ApiHubEvent
```

Comment:
Just noticed that the hub event model is not being followed properly here. We should have something like this:

```dart
// api_hub_event.dart
// Modified slightly - don't think we need the base ApiHubEventPayload class.
class ApiHubEvent<HubEventPayload> extends HubEvent<HubEventPayload> {
  ApiHubEvent(
    super.eventName, {
    super.payload,
  });
}

// api_subscription_hub_event.dart
class SubscriptionHubEvent extends ApiHubEvent<SubscriptionDetails> {
  SubscriptionHubEvent._(this.details) : super(_name, payload: details);
}

```